stumpy/mstump.py

```diff
@@ -525,13 +569,32 @@ def multi_distance_profile(
         and the Euclidean distance, respectively. This parameter is ignored when
         `normalize == True`.
 
+    T_subseq_isconstant : numpy.ndarray or callable or list, default None
+        A parameter that is used to show whether a subsequence of a time series in `T`
+        is constant(True) or not. T_subseq_isconstant can be a 2D boolean numpy.ndarry
+        or it be a callable function, which will be applied to each time series in `T`.
+        It can also be list of length `T.shape[0]`, i.e. dimension size of `T`. In this
+        case, T_subseq_isconstant[i] corresponds to the i-th time series `T[i]`. Each
+        element of list can be a 1D boolean np.ndarray, a callable function, or None.
+
     Returns
     -------
     D : numpy.ndarray
         Multi-dimensional distance profile for the window with index equal to
         `query_idx`
     """
-    T, M_T, Σ_T, T_subseq_isconstant = core.preprocess(T, m)
+    if T_subseq_isconstant is None or callable(T_subseq_isconstant):
+        T_subseq_isconstant = [T_subseq_isconstant] * T.shape[0]
```

Comment:
Yeah. In fact, I think it SHOULD BE POSSIBLE to just pass `None` or function. However, since I wanted to handle list (containing a combination of 1D boolean array, function, `None`), I decided to just repeat them and make a list first. And, finally just use the list.